index.js

```diff
@@ -16,6 +16,139 @@ cookieButton.addEventListener('click', () => {
      cookie.classList.add('hidden');
 })
 
+//active navbar elements
+
+function activateMenuLink(allLinks, linkToActivate) { //funkcja dodajaca active do linkow  (pierwszy parametr lista wszystkich linkow, w drugim link ktory chce aktywowac)
+     for(let i = 0; i < allLinks.length; i++){ //petla dla calej listy linkow
+
+          let link = allLinks[i]; // element w tablicy all links
+
+          if (link.classList.contains('active')) { //  jesli jest active to usuwamy
+               link.classList.remove("active");
+          }
+     }
+
+     if (!linkToActivate.classList.contains('active')) { // jesli nie zawiera active to dodajemy
+          linkToActivate.classList.add("active");
+     }
+
+     let href = linkToActivate.getAttribute("href"); //pobieramy atrybut href (np #features) 
+
+     window.history.pushState(href, "Vitamina - "+href, href); // dodajemy go to historii przegladarki- zmienia sie url bez reloadu
+}
+
+const navigationMenuContainer = document.querySelector(".navigation");
+const navigationMenuElement = document.querySelector(".navigation__menu__list");
+const navigationMenuElementFocus = document.querySelectorAll(".navigation__menu__list__link");
+
+for (let i = 0; i < navigationMenuElementFocus.length; i++) {
+          navigationMenuElementFocus[i].addEventListener("click", (event) => {
+          activateMenuLink(navigationMenuElementFocus, event.target); //wywolujemy funkcje na kliku
+     });
+}
+
+// scroll handler
+
+let linksDestination = document.getElementsByClassName("linkLoc"); // zbieramy elementy majace klase linkLoc w HTML
+let links = document.getElementsByClassName("navigation__menu__list__link"); 
+let positionsList = []; // tworzymy pusta liste 
+let linkHrefs = []; // tworzymy pusta liste hfefow
+
+for(let i = 0; i < links.length; i++){
+     let link = links[i]; // sprawdzamy kazdy element
+     linkHrefs.push(link.getAttribute("href")); //  pushujemy hrefy do pustej listy 
+}
+for(let i = 0; i < linksDestination.length; i++){ // petla sprawdzajaca elementy z klasa linkLoc
+     let linkDestination = linksDestination[i];  //sekcje docelowych miejsc do ktorych przekierowuja linki w navie
+
+     positionsList.push(linkDestination.offsetTop); //  ile pikseli od gory dokumentu zaczyna sie ta sekcja
+}
+
+positionsList.push(document.body.clientHeight); //pushujemy cala wysokosc widoku, zeby ostatni element mial wartosc
+
+let pairListStart = []; // pusta lista- poczatkowe pozycje elementow
+let pairListEnd = []; // pusta liosta- koncowe pozycje elementow
+
+// 200,500,600,2000, 3500
+
+
+for(let i =0; i < positionsList.length-1;i++) // petla do generowania par pozycji
```

Comment:
nie bardzo łapie co te funkcje maja robic, moze lepiej je wydzielic do jakiejs klasy, obiektu, zeby to jakos opakowac jakas nazwa? Bo nawet komentarze mi jakos specjalnie nie pomagaja